daml-script/daml/Daml/Script.daml

```diff
@@ -562,7 +562,8 @@ data Script a = Script with
   deriving Functor
 
 instance CanAbort Script where
-  abort s = Script (\_ -> error s) ()
+  -- abort s = Script (\_ -> error s) ()
+  abort s = throw (GeneralError s)
```

Comment:
IIUC, the weird behavior was because `error s` in the old version was at type `Free ScriptF ((),())`, so the abstract script command itself was an error (rather than a command generating the error).

Reply:
this means however that the version of the problematic code with `error` instead of `abort` still fails, and I'm not sure how to fix it or if it can be fixed at all with the current model

Reply:
Can you help me understand how this produced the error? If we look at the trace statements we can see that first we see no exception and then we get to the `error/abort/…` call that throws one but this seems to be caught within the `try` still. To me that looks more like the `try` scoping is off?